src/rules/no-unnormalized-keys.js

```diff
@@ -3,7 +3,19 @@
  * @author Bradley Meck Farias
  */
 
-export default {
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/** @typedef {"unnormalizedKey"} NoUnnormalizedKeysMessageIds */
+/** @typedef {import("../types.ts").JSONRuleDefinition<[], NoUnnormalizedKeysMessageIds>} NoUnnormalizedKeysRuleDefinition */
```

Comment:
```suggestion
/** @typedef {import("../types.ts").JSONRuleDefinition<[{form:string}], NoUnnormalizedKeysMessageIds>} NoUnnormalizedKeysRuleDefinition */
```

I think we can specify the type for the rule options like this, so that `context.options` gets the expected type, and then we wouldn't need the type annotation `/** @type {{form:string}[]} */` below.

Reply:
Ah that is brilliant, thanks!